File: iRead/iReader/Base/Common/IRBaseViewController/IRBaseViewController.m
//
//  IRBaseViewController.m
//  iReader
//
//  Created by zzyong on 2018/9/25.
//  Copyright © 2018年 zouzhiyong. All rights reserved.
//

#import "IRBaseViewController.h"

@interface IRBaseViewController ()

@end

@implementation IRBaseViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    IRDebugLog(@"[%@ viewDidLoad]", NSStringFromClass(self.class));
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    IRDebugLog(@"[%@ viewWillAppear]", NSStringFromClass(self.class));
}

- (void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
    
    IRDebugLog(@"[%@ viewWillDisappear]", NSStringFromClass(self.class));
}

#pragma mark -

- (void)setupLeftBackBarButton
{
    UIBarButtonItem *back = [[UIBarButtonItem alloc] initWithImage:[[UIImage imageNamed:@"arrow_back"] imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal]
                                                             style:UIBarButtonItemStylePlain
                                                            target:self
                                                            action:@selector(onClickedLeftBackItem:)];
    self.navigationItem.leftBarButtonItem = back;
}

- (void)onClickedLeftBackItem:(UIBarButtonItem *)item
{
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: iRead/IRCommonLib/IRCommonLib/IRBaseViewcontroller/IRBaseViewcontroller.swift
//
//  IRBaseViewcontroller.swift
//  IRCommonLib
//
//  Created by zzyong on 2020/9/24.
//  Copyright © 2020 zzyong. All rights reserved.
//

import UIKit

open class IRBaseViewcontroller: UIViewController {

    open override func viewDidLoad() {
        super.viewDidLoad()
        IRDebugLog(self)
    }

    open override func viewWillAppear(_ animated: Bool) {
        IRDebugLog(self)
        return super.viewWillAppear(animated)
    }
    
    open override func viewDidDisappear(_ animated: Bool) {
        IRDebugLog(self)
        return super.viewDidDisappear(animated)
    }
    
    open func setupLeftBackBarButton() {
        let backImg = UIImage.init(named: "arrow_back")?.withRenderingMode(.alwaysOriginal)
        let back = UIBarButtonItem.init(image: backImg, style: .plain, target: self, action: #selector(didClickedLeftBackItem(item:)))
        self.navigationItem.leftBarButtonItem = back
    }
    
    @objc private func didClickedLeftBackItem(item: UIBarButtonItem) {
        self.navigationController?.popViewController(animated: true)
    }
}
